src/Servers/HttpSys/src/AsyncAcceptContext.cs

```diff
@@ -62,51 +68,38 @@ internal ValueTask<RequestContext> AcceptAsync()
         {
-            try
+            // Handle any errors
+            if (errorCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS &&
+                errorCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_MORE_DATA)
             {
-                if (errorCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS &&
-                    errorCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_MORE_DATA)
-                {
-                    _mrvts.SetException(new HttpSysException((int)errorCode));
-                    return;
-                }
+                _mrvts.SetException(new HttpSysException((int)errorCode));
+            }
+            else if (errorCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS)
+            {
+                var requestContext = _requestContext;
+                // It's important that we clear the request context before we set the result
+                // we want to reuse the acceptContext object for future accepts.
+                _requestContext = null;
 
-                if (errorCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS)
-                {
-                    var requestContext = _requestContext;
-                    // It's important that we clear the request context before we set the result
-                    // we want to reuse the acceptContext object for future accepts.
-                    _requestContext = null;
+                _mrvts.SetResult(requestContext);
+            }
+            else
+            {
+                Debug.Assert(errorCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_MORE_DATA, $"Making another request and the error code is not ERROR_MORE_DATA, {errorCode}");
 
-                    _mrvts.SetResult(requestContext);
-                }
-                else
+                try
                 {
-                    //  (uint)backingBuffer.Length - AlignmentPadding
+                    // We need to issue a new request because our buffer was too small the first time.
                     AllocateNativeRequest(numBytes, _requestContext.RequestId);
 
-                    // We need to issue a new request, either because auth failed, or because our buffer was too small the first time.
-                    uint statusCode = QueueBeginGetContext();
-
-                    if (statusCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS &&
-                        statusCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_IO_PENDING)
-                    {
-                        // someother bad error, possible(?) return values are:
-                        // ERROR_INVALID_HANDLE, ERROR_INSUFFICIENT_BUFFER, ERROR_OPERATION_ABORTED
-                        _mrvts.SetException(new HttpSysException((int)statusCode));
-                    }
+                    // Ignore the result here since the both sync and async cases should be handled by the call
+                    _ = QueueBeginGetContext();
```

Comment:
Is there a test for a large request that hits this code path?

Reply:
Can we assert `statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS`?

Reply:
> Is there a test for a large request that hits this code path?

I'll check

> Can we assert statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS?

Yes